src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -157,6 +162,10 @@ RTPSParticipantImpl::RTPSParticipantImpl(
         m_network_Factory.RegisterTransport(&descriptor, &m_att.properties);
 
 #ifdef SHM_TRANSPORT_BUILTIN
+#ifdef ANDROID
+        if (access(BOOST_INTERPROCESS_SHARED_DIR_PATH, W_OK) == F_OK)
```

Comment:
I would move this check to `SharedMemTransport::init`. This way we avoid including `boostconfig.hpp` on this source file.